detect_secrets/core/audit.py

```diff
@@ -612,36 +602,16 @@ def get_raw_secret_value(
         plugin_settings,
     )
 
-    for raw_secret in raw_secret_generator(
-        plugin,
-        secret_line,
-        filetype=determine_file_type(filename),
-    ):
-        secret_obj = PotentialSecret(
-            plugin.secret_type,
-            filename,
-            secret=raw_secret,
-        )
+    with codecs.open(filename, encoding='utf-8') as f:
+        plugin_secrets = plugin.analyze(f, filename)
```

Comment:
This looks like we're re-scanning the file **per secret audited**. How does this perform with large files?

Reply:
It seems like we can avoid this by either changing or not using `_secret_generator` before calling this method

https://github.com/Yelp/detect-secrets/blob/2b40c99e4d5282102ec34d45dac9bd7884ebcd64/detect_secrets/core/audit.py#L288-L292

Right now we call this method on every secret, when we only need to call it for each key in `baseline['results']`.